pkg/testworkflows/testworkflowprocessor/action/group.go

```diff
@@ -0,0 +1,74 @@
+package action
+
+import (
+	"slices"
+
+	"github.com/kubeshop/testkube/pkg/testworkflows/testworkflowprocessor/action/actiontypes"
+)
+
+func Group(actions []actiontypes.Action) (groups [][]actiontypes.Action) {
+	// Detect "start" and "execute" instructions
+	startIndexes := make([]int, 0)
+	startInstructions := make(map[string]int)
+	containerInstructions := make(map[string]int)
+	executeInstructions := make(map[string]int)
+	executeIndexes := make([]int, 0)
+	for i := range actions {
+		if actions[i].Start != nil {
+			startInstructions[*actions[i].Start] = i
+			startIndexes = append(startIndexes, i)
+		} else if actions[i].Execute != nil {
+			executeInstructions[actions[i].Execute.Ref] = i
+			executeIndexes = append(executeIndexes, i)
+		} else if actions[i].Container != nil {
+			containerInstructions[actions[i].Container.Ref] = i
+		} else if actions[i].Setup != nil {
+			executeIndexes = append(executeIndexes, i)
+		}
+	}
+
+	// Start from end, to fill as much as it's possible
+	slices.Reverse(executeIndexes)
+	slices.Reverse(startIndexes)
+
+	// Fast-track when there is only a single instruction to execute
+	if len(executeIndexes) <= 1 {
+		return [][]actiontypes.Action{actions}
+	}
+
+	// Basic behavior: split based on each execute instruction
+	for _, executeIndex := range executeIndexes {
+		if actions[executeIndex].Setup != nil {
+			groups = append([][]actiontypes.Action{actions[executeIndex:]}, groups...)
+			actions = actions[:executeIndex]
+			continue
+		}
+		ref := actions[executeIndex].Execute.Ref
+		startIndex := startInstructions[ref]
+		if containerIndex, ok := containerInstructions[ref]; ok && containerIndex < startIndex {
+			startIndex = containerIndex
+		}
+
+		//// TODO: Combine multiple operations in a single container if it's possible
+		//if i != len(executeIndexes)-1 {
```

Comment:
should it be uncommented or removed?

Reply:
Either removed or left - it's example how to perform combining multiple operations, yet it has tons of edge cases